core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -2897,6 +2895,31 @@ private void optimiseOrderBy(QueryModel model, int topLevelOrderByMnemonic) {
         }
     }
 
+    private boolean canPushDownOrderByAdvice(QueryModel model, ObjList<ExpressionNode> orderByAdvice) {
+        if (model.getGroupBy().size() > 0) {
+            return false;
+        }
+        if (model.getSampleBy() != null) {
+            // The ORDER BY advice cannot be pushed down through SAMPLE BY, because SAMPLE BY
+            // requires designated timestamps to be sorted by their natural (ascending) order.
+            return false;
+        }
+        if (model.getSelectModelType() == QueryModel.SELECT_MODEL_DISTINCT) {
+            // According to the logical order, the ORDER BY clause should always be applied after DISTINCT.
+            // However, here, we specifically address the case where the ORDER BY advice contains only
+            // the designated timestamp. This case is handled by the DistinctTimeSeriesRecordCursorFactory,
+            // which does not expose information about following the advice to the factories relying on it
+            // (so it's up to them to handle the ORDER BY clause). It only exposes the scanning direction,
+            // controlled by the advice. If the scanning direction aligns with the ORDER BY clause and the
```

Comment:
nit: I guess `the ORDER BY clause and the clause` should be `the ORDER BY clause and the DISTINCT clause`

Reply:
Not really. What I meant is that the ORDER BY clause has to include only the timestamp, and the order must match the scanning direction.

I've changed the wording.

Reply:
Reads better now, thanks!